src/init.cpp

```diff
@@ -1587,8 +1587,7 @@ bool AppInitMain()
 
     // ********************************************************* Step 8: start indexers
     if (gArgs.GetBoolArg("-txindex", DEFAULT_TXINDEX)) {
-        auto txindex_db = MakeUnique<TxIndexDB>(nTxIndexCache, false, fReindex);
-        g_txindex = MakeUnique<TxIndex>(std::move(txindex_db));
+        g_txindex = MakeUnique<TxIndex>(nTxIndexCache, false, fReindex);
```

Comment:
Why is this being changed?

Reply:
I think it makes sense for the DB to be encapsulated in the index interface, even if it is implemented separately internally.

Reply:
Agree with being encapsulated. It's just a unrelated change AFAIU, but a nice to include.